src/plugin/protocol-selenium/src/java/org/apache/nutch/protocol/selenium/HttpResponse.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nutch.protocol.selenium;
+
+import com.google.common.base.Predicate;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.metadata.SpellCheckedMetadata;
+import org.apache.nutch.net.protocols.Response;
+import org.apache.nutch.storage.WebPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+
+public class HttpResponse implements Response {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HttpResponse.class);
+    private Http http;
+    private URL url;
+    private byte[] content;
+    private int code;
+    private Metadata headers = new SpellCheckedMetadata();
+
+    /**
+     * The nutch configuration
+     */
+    private Configuration conf = null;
+
+    public HttpResponse(Http http, URL url, WebPage page, Configuration conf) throws UnsupportedEncodingException {
+
+        this.conf = conf;
+        this.http = http;
+        this.url = url;
+
```

Comment:
TODO: Use a HTTPUrlConnection to make a HEAD request querying for content type before opening in Selenium?

Should we attempt to load another fetcher such as protocol-httpclient if the file should not be handled by Selenium?